src/main/java/address/controller/MainController.java

```diff
@@ -237,6 +241,39 @@ private FXMLLoader loadFxml(String fxmlResourcePath) {
         }
     }
 
+    public Optional<List<Tag>> getPersonsTagsInput(List<ReadOnlyViewablePerson> persons) {
+        FXMLLoader loader = new FXMLLoader();
+        loader.setLocation(MainApp.class.getResource(FXML_TAG_SELECTION_EDIT_DIALOG));
+        try {
+            AnchorPane pane = loader.load();
+
+            // Create the dialog Stage.
+            Stage dialogStage = new Stage();
+            dialogStage.initModality(Modality.WINDOW_MODAL);
+            dialogStage.initOwner(primaryStage);
+            dialogStage.initStyle(StageStyle.TRANSPARENT);
+
+            Scene scene = new Scene(pane, Color.TRANSPARENT);
+            dialogStage.setScene(scene);
+
+            TagSelectionEditDialogController controller = loader.getController();
+            controller.setTags(modelManager.getTagsAsReadOnlyObservableList(),
+                               ReadOnlyPerson.getCommonTags(persons));
+            controller.setDialogStage(dialogStage);
+
+            dialogStage.showAndWait();
+
+            if (controller.isOkClicked()) {
+                return Optional.of(controller.getFinalAssignedTags());
+            }
+
+        } catch (IOException e) {
+            logger.warn("Error launching tag selection dialog: {}", e);
+            assert false : "Error loading fxml : " + FXML_TAG_SELECTION_EDIT_DIALOG;
+        }
+        return Optional.empty();
+    }
```

Comment:
this method is to get the final tags to be updated. if an empty list is returned. the person tags will be empty. where an empty optional indicates no changes to be made